### ISSUE TITLE
MSW: overrideResponse can be a function

### ISSUE THREAD

**Issue by purefun** [anymaniax/orval#1329]
Currently, a msw mock handler would be:

```ts
import { HttpResponse, HttpResponseResolver, delay, http } from 'msw'

export const getSomethingMockHandler = (overrideResponse?: SomethingResponse) => {
  return http.get('/something', async () => {
    await delay(1000);
    return new HttpResponse(JSON.stringify(overrideResponse ? overrideResponse : getSomethingResponseMock()),
      {
        status: 200,
        headers: {
          'Content-Type': 'application/json',
        }
      }
    )
  })
}

```

But it would be nice if overrideResponse is a function. So that we can produce response programmatically.

```ts
export const getSomethingMockHandler = (overrideResponse?: (resolver: HttpResponseResolver) => SomethingResponse | Promise<SomethingResponse>) => {
  return http.get('/something', async (resolver) => {
    await delay(1000);
    const response = overrideResponse ? overrideResponse(resolver) : getSomethingResponseMock()
    return new HttpResponse(JSON.stringify(response)),
      {
        status: 200,
        headers: {
          'Content-Type': 'application/json',
        }
      }
    )
  })
}

```

**Comment by melloware** [comment]
@soartec-lab is this the same issue you just fixed in 6.28.0?   https://github.com/anymaniax/orval/issues/1293

**Comment by soartec-lab** [comment]
@melloware 
No, this is separate from #1293. I'll add an enhancement label to this.
